kroxylicious-sample/src/main/java/io/kroxylicious/sample/util/SampleFilterTransformer.java

```diff
@@ -46,7 +46,7 @@ public static void transform(ProduceRequestData.PartitionProduceData partitionDa
      * @param config the transform configuration
      */
     public static void transform(FetchResponseData.PartitionData partitionData, KrpcFilterContext context, SampleFilterConfig config) {
-        partitionData.setRecords(transformPartitionRecords((MemoryRecords) partitionData.records(), context, config.getFindValue(), config.getReplacementValue()));
+        partitionData.setRecords(transformPartitionRecords((AbstractRecords) partitionData.records(), context, config.getFindValue(), config.getReplacementValue()));
```

Comment:
Having to cast to AbstractRecords is a bit of code smell. Working with Interfaces is usually prefered as this should represent stable API rather than implementation.  I think you can write `records.firstBatch()` in terms of the Records API by using `records.batchIterator()`.

Reply:
Casting to `AbstractRecords` is in my opinion not much different from casting to `MemoryRecords` (if anything, it's safer). I see what you're saying about using `Records` instead and I don't necessarily disagree, but I think this could perhaps be part of a wider discussion around how we should be handling batching and other related records-handling problems, rather than as part of this PR.

At the moment what we're doing here is compliant with what we're doing elsewhere in Kroxylicious, so I'm of the opinion that for this specific PR using AbstractRecords should be fine.

Reply:
Ok, happy to defer.   Kroxylicious implementation should be favouring programming against interface too of course!